src/sentry/static/sentry/app/views/settings/components/settingsLayout.tsx

```diff
@@ -14,18 +19,88 @@ type Props = {
   children: React.ReactNode;
 } & RouteComponentProps<{}, {}>;
 
-function SettingsLayout(props: Props) {
-  const {params, routes, route, router, renderNavigation, children} = props;
+type State = {
+  navVisible: boolean;
+  navOffsetTop: number;
+};
+
+class SettingsLayout extends React.Component<Props, State> {
+  static propTypes = {
+    renderNavigation: PropTypes.func,
+    route: PropTypes.object,
+    router: PropTypes.object,
+    routes: PropTypes.array,
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.headerRef = React.createRef();
 
-  // We want child's view's props
-  const childProps = children && React.isValidElement(children) ? children.props : props;
-  const childRoutes = childProps.routes || routes || [];
-  const childRoute = childProps.route || route || {};
-  return (
-    <React.Fragment>
+    // Close the navigation when navigating.
+    this.unlisten = browserHistory.listen(() => this.toggleNav(false));
```

Comment:
wouldn't be better to move it to the componentDidMount? it's recommended to avoid introducing any side-effects or subscriptions in the constructor